### DISCUSSION TITLE
Snippet: How to add custom blocks to Navigation menus

### DISCUSSION THREAD

**Discussion by ndiego** [WordPress/developer-blog-content#355] [Topic Ideas]
Adding custom blocks to the Navigation block is a common need, such as blocks for icons or a mega menu. By default, the Navigation block restricts the blocks that can be used within it. However, with a bit of code, you can enable your custom block to be used in Navigation. This snippet will show you how.

This idea originated from [this discussion](https://wordpress.slack.com/archives/C01KDAZJMQ9/p1733663708006679) in WP Slack.

**Comment by ndiego** [comment]
In preparation for this snippet (hopefully) being approved. I have started an initial draft: https://docs.google.com/document/d/1d9_Emzx-zVQn6YFS6A7xTru500y-oqs6yqa85jjvQQI/edit?tab=t.0

**Comment by bph** [comment]
Topic approved. Conversation continues on the issue
- #365

**Comment by bph** [comment]
Published: 
[Snippet: How to add custom blocks to navigation menus](https://developer.wordpress.org/news/snippets/how-to-add-custom-blocks-to-navigation-menus/)
